src/views/Generator/RuleEditor/ParameterizationEditor/FileSelect.tsx

```diff
@@ -0,0 +1,85 @@
+import { Code } from '@radix-ui/themes'
+import { ControlledSelect, FieldGroup } from '@/components/Form'
+import { ParameterizationRule } from '@/types/rules'
+import { useFormContext } from 'react-hook-form'
+import { useGeneratorStore } from '@/store/generator'
+import { useMemo } from 'react'
+import { useDataFilePreview } from '@/views/DataFile/DataFile.hooks'
+
+export function FileSelect() {
+  const {
+    control,
+    watch,
+    formState: { errors },
+  } = useFormContext<ParameterizationRule>()
+
+  const fileName = watch('value.fileName')
+  const propertyName = watch('value.propertyName')
+  const files = useGeneratorStore((store) => store.files)
+  const { data: preview, isLoading } = useDataFilePreview(fileName)
+
+  const fileOptions = useMemo(() => {
+    return files.map((file) => ({
+      value: file.name,
+      label: (
+        <Code size="2" truncate variant="ghost">
+          {file.name}
+        </Code>
+      ),
+    }))
+  }, [files])
+
+  const propsOptions = useMemo(() => {
+    return preview?.props.map((prop) => ({
+      value: prop,
+      label: (
+        <Code size="2" truncate variant="ghost">
+          {prop}
+        </Code>
+      ),
+    }))
+  }, [preview])
+
+  return (
+    <>
+      <FieldGroup name="value.fileName" errors={errors} label="Data file">
+        <ControlledSelect
+          options={fileOptions}
+          control={control}
+          name="value.fileName"
+          selectProps={{
+            // Automatically open the select when switching to data file value type
+            // in new parameterization rule
+            defaultOpen: !fileName,
+          }}
+          contentProps={{
+            css: { maxWidth: 'var(--radix-select-trigger-width)' },
+            position: 'popper',
+          }}
+        />
+      </FieldGroup>
+      <FieldGroup
+        name="value.fileName"
+        errors={errors}
+        label="Key"
+        hint="Used to retrieve a value from a randomly selected row in the data file"
```

Comment:
Does this make sense? Alternatives I had in mind were `Property` or even `Property`/`Column` depending on the file type (although I'd prefer to treat all data files types the same)

Reply:
"Column" would be clearer IMO, because that's how the data is typically organized in CSVs, which makes it a more intuitive term for users to understand. "Key" might be a bit ambiguous 🤔

Reply:
I agree that it's rather ambiguous. But then "Column" might not be the right fit for JSON files